core/src/storage2/lazy/lazy_cell.rs

```diff
@@ -307,6 +307,34 @@ where
     pub fn get_mut(&mut self) -> Option<&mut T> {
         self.load_entry_mut().value_mut().into()
     }
+
+    /// Sets the value in this cell to `value`, without executing any reads.
+    ///
+    /// # Note
+    ///
+    /// No reads from contract storage will be executed.
+    ///
+    /// This method should be preferred over dereferencing or `get_mut`
+    /// in case the returned value is of no interest to the caller.
+    ///
+    /// # Panics
+    ///
+    /// If accessing the inner value fails.
+    pub fn set(&mut self, new_value: T) {
+        // SAFETY: This is critical because we mutably access the entry.
+        let cache = unsafe { &mut *self.cache.get_ptr().as_ptr() };
+        if let Some(cache) = cache.as_mut() {
+            //  Cache is already populated we simply overwrite its already existing value.
+            cache.put(Some(new_value));
```

Comment:
One difference that could have a performance impact is that we are using `Entry::put` here but `Entry::value_mut` + assign in `get_mut`. Maybe the latter is more efficient?

Reply:
Good catch, but I tried it out and it actually slows `set` down :/.